parm/config/config.ufs

```diff
@@ -0,0 +1,356 @@
+    "C768")
+        export DELTIM=150
+        export layout_x=8
+        export layout_y=12
+        export layout_x_gfs=16
+        export layout_y_gfs=12
+        export nth_fv3=4
+        export nth_fv3_gfs=4
+        export cdmbgwd="4.0,0.15,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+        export WRITE_GROUP=2
+        export WRTTASK_PER_GROUP=64
+        export WRITE_GROUP_GFS=4
+        export WRTTASK_PER_GROUP_GFS=64
+        export WRTIOBUF="32M"
+        ;;
+    "C1152")
+        export DELTIM=120
+        export layout_x=8
+        export layout_y=16
+        export layout_x_gfs=8
+        export layout_y_gfs=16
+        export nth_fv3=4
+        export nth_fv3_gfs=4
+        export cdmbgwd="4.0,0.10,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+        export WRITE_GROUP=4
+        export WRTTASK_PER_GROUP=$(echo "2*${npe_node_max}" |bc)  # TODO: this is pointless because below this is getting reset
+        export WRITE_GROUP_GFS=4
+        export WRTTASK_PER_GROUP_GFS=$(echo "2*${npe_node_max}" |bc)  # TODO: this is pointless because below this is getting reset
+        export WRTIOBUF="48M"
+        ;;
+    "C3072")
+        export DELTIM=90
+        export layout_x=16
+        export layout_y=32
+        export layout_x_gfs=16
+        export layout_y_gfs=32
+        export nth_fv3=4
+        export nth_fv3_gfs=4
+        export cdmbgwd="4.0,0.05,1.0,1.0"  # mountain blocking, ogwd, cgwd, cgwd src scaling
+        export WRITE_GROUP=4
+        export WRTTASK_PER_GROUP=$(echo "3*${npe_node_max}" |bc)  # TODO: this is pointless because below this is getting reset
+        export WRITE_GROUP_GFS=4
+        export WRTTASK_PER_GROUP_GFS=$(echo "3*${npe_node_max}" |bc)  # TODO: this is pointless because below this is getting reset
+        export WRTIOBUF="64M"
+        ;;
+    *)
+        echo "FATAL ERROR: Unsupported FV3 resolution = ${fv3_res}, ABORT!"
+        exit 1
+        ;;
+esac
+
+if [[ "${WRTTASK_PER_GROUP}" -gt "${npe_node_max}" ]]; then export WRTTASK_PER_GROUP=${npe_node_max} ; fi
+if [[ "${WRTTASK_PER_GROUP_GFS}" -gt "${npe_node_max}" ]]; then export WRTTASK_PER_GROUP_GFS=${npe_node_max} ; fi
+
+(( ntasks_fv3 = layout_x * layout_y * 6 ))
+(( ntasks_fv3_gfs = layout_x_gfs * layout_y_gfs * 6 ))
+export ntasks_fv3
+export ntasks_fv3_gfs
+
+(( ntasks_quilt = WRITE_GROUP * WRTTASK_PER_GROUP ))
+(( ntasks_quilt_gfs = WRITE_GROUP_GFS * WRTTASK_PER_GROUP_GFS ))
+export ntasks_quilt
+export ntasks_quilt_gfs
+
+# Determine whether to use parallel NetCDF based on resolution
+case ${fv3_res} in
+  "C48" | "C96" | "C192" | "C384")
+    export OUTPUT_FILETYPE_ATM="netcdf"
+    export OUTPUT_FILETYPE_SFC="netcdf"
+    ;;
+  "C768" | "C1152" | "C3072")
+    export OUTPUT_FILETYPE_ATM="netcdf_parallel"
+    export OUTPUT_FILETYPE_SFC="netcdf_parallel"
+    ;;
+esac
+
+# MOM6 specific settings
+if [[ "${skip_mom6}" == "false" ]]; then
+  export nth_mom6=1  # MOM6 is not thread-safe (allegedly)
```

Comment:
i don't know if its that it's not thread safe i think it's that you don't get advantage using threads

Reply:
I will update the comment appropriately.